pkg/sink/interceptors_test.go

```diff
@@ -0,0 +1,168 @@
+			APIVersion: "v1",
+		},
+		expected: "http://foo.default.svc/",
+		wantErr:  false,
+	}, {
+		name: "non services",
+		ref: corev1.ObjectReference{
+			Kind:       "Blah",
+			Name:       "foo",
+			APIVersion: "v1",
+		},
+		expected: "",
+		wantErr:  true,
+	}}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			url, err := GetURI(&tc.ref, eventListenerNs)
+			if err != nil {
+				if !tc.wantErr {
+					t.Errorf("Unexpected error: %v", err)
+				}
+			} else if diff := cmp.Diff(tc.expected, url.String()); diff != "" {
+				t.Errorf("Did not get expected URL: %s", diff)
+			}
+		})
+	}
+}
+
+func TestCreateOutgoingRequest(t *testing.T) {
+	reqBody, _ := json.Marshal(map[string]string{
+		"eventType": "push",
+		"foo":       "bar",
+	})
+
+	req, _ := http.NewRequest(http.MethodPost, "http://event.listener.url", ioutil.NopCloser(bytes.NewBuffer(reqBody)))
+	req.Header.Add("Content-type", "application/json")
+	req.Header.Add("X-Event-Id", "blah")
+	eventProcessorUrl, _ := url.Parse("http://some.other.url")
+
+	expectedReq, _ := http.NewRequest(http.MethodPost, "http://some.other.url", ioutil.NopCloser(bytes.NewBuffer(reqBody)))
+	expectedReq.Header.Add("Content-type", "application/json")
+	expectedReq.Header.Add("X-Event-Id", "blah")
+
+	outgoing := createOutgoingRequest(context.Background(), req, eventProcessorUrl, reqBody)
+
+	respBody, err := ioutil.ReadAll(outgoing.Body)
+	if err != nil {
+		t.Errorf("Failed to parse outgoing request body: %q", err)
+	}
+
+	if outgoing.URL != eventProcessorUrl {
+		t.Errorf("Outgoing request has unexpected URL: %s", outgoing.URL)
+	}
+
+	if diff := cmp.Diff(expectedReq.Header, outgoing.Header); diff != "" {
+		t.Errorf("Did not create request with expected headers: %s", diff)
+	}
+
+	if diff := cmp.Diff(reqBody, respBody, cmpopts.IgnoreUnexported()); diff != "" {
+		t.Errorf("Did not create request with expected body: %s", diff)
+	}
+}
+
+func TestMakeRequest(t *testing.T) {
+	reqBody, _ := json.Marshal(map[string]string{
+		"eventType": "push",
+		"foo":       "bar",
+	})
+
+	tcs := []struct {
+		name            string
+		handler         http.HandlerFunc
+		expectedPayload []byte
+		wantErr         bool
+	}{{
+		name: "status 200",
+		handler: func(w http.ResponseWriter, r *http.Request) {
+			p, _ := ioutil.ReadAll(r.Body)
+			_, _ = w.Write(p)
+		},
+		expectedPayload: reqBody,
+		wantErr:         false,
+	}, {
+		name: "status 400",
+		handler: func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusBadRequest)
+		},
+		wantErr: true,
+	}}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(tc.handler))
+			defer ts.Close()
+
+			req, err := http.NewRequest(http.MethodPost, ts.URL, bytes.NewBuffer(reqBody))
+			if err != nil {
+				t.Fatalf("Error trying to create request: %q", err)
+			}
+			payload, err := makeRequest(http.DefaultClient, req)
+			expectedPayload := tc.expectedPayload
```

Comment:
Could this check the status code rather than the body? It makes sense that the body is not the one we expect in this test, but that should be the source of truth?

Reply:
Yes, makeRequest returns an error if the status code is not 200. The test case tests that it does indeed return an error if the service returns a 400 status code.